sql/src/main/java/org/apache/druid/sql/calcite/rule/DruidJoinRule.java

```diff
@@ -74,7 +73,7 @@ private DruidJoinRule(final PlannerContext plannerContext)
             operand(DruidRel.class, any())
         )
     );
-    this.enableLeftScanDirect = QueryContexts.getEnableJoinLeftScanDirect(plannerContext.getQueryContext());
```

Comment:
`isEnableJoinLeftScanDirect` seems too specific a use case to be a part of the base `QueryContext` class itself. Why have we moved it from `QueryContexts`?

Reply:
IMO, now `QueryContexts` should remain just as a holder of predefined context parameters. It doesn't give any more benefit over `QueryContext`. `getEnableJoinLeftScanDirect` is just a util method for our convenience and they all should be moved to `QueryContext`.

Reply:
👍🏻 Moving all of these methods to `QueryContext` would be much better than it is now.